src/data/CHDatasource.ts

```diff
@@ -218,6 +218,11 @@ export class Datasource
     return {
       ...query,
       rawSql: this.replace(rawQuery, scoped) || '',
+      customSettings: this.settings.jsonData.customSettings?.map(({ setting, value }) => {
+        return {
+          setting, value: this.replace(this.applyConditionalAll(value, getTemplateSrv().getVariables()), scoped) || ''
```

Comment:
I am not sure if we should evaluate custom template variables. It would be a confusing experience to have your custom settings work for a dashboard and error in Explore view, for example.

Reply:
Agreed that it's not a perfect situation, however there is a use-case here for passing user information as a parameter (ROW POLICY often used custom settings). Maybe we could pass this some other way? Evaluating the templates on the server side for a subset of variables?

Reply:
Ok, for now I decided to simplify this PR and remove this logic. Templated settings can be passed in individual queries like this for now:

```
SELECT * FROM LS_DEMO
SETTINGS ls_profile='${__user.login}';
```

This is just a minor inconvenience for now

Reply:
I now have a fork of `sqlds` and `clickhouse-datasource` that interpolates server-side macros in custom settings, and adds a new macro for `$__userLogin`.

- https://github.com/aramperes/sqlds/compare/main...aramperes:sqlds:user-login-macro
- https://github.com/aramperes/grafana-clickhouse-datasource/compare/custom-clickhouse-settings...aramperes:grafana-clickhouse-datasource:user-macro